extension_service/datastore/providers/firestore_test.py

```diff
@@ -0,0 +1,317 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime
+from typing import Dict
+
+from google.cloud import firestore
+from google.cloud.firestore_v1.base_query import FieldFilter
+
+import models
+
+from . import firestore as firestore_provider
+
+
+class MockDocument(Dict):
```

Comment:
I think maybe we don't want to use Mock's here, since we are trying to move away from Mocks in Postgres. Maybe @Yuan325 can weigh in on what she'd prefer?

Reply:
Yeah, most likely it will be re-written after the integration test is implemented.

Reply:
I'll leave it for now and re-write it later then.